Sources/StringsLintFramework/Rules/Lint/StructuredPlaceholderCommentRule.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  StructuredPlaceholderCommentRule.swift
+//  StringsLintFramework
+//
+//  Created by Mark Hall on 2022-07-18.
+//
+
+import Foundation
+
+public class StructuredPlaceholderCommentRule {
+    private var declaredStrings = [LocalizedString]()
+    var severity: ViolationSeverity
+
+    private let declareParser: LocalizableParser
+
+    private var validPlaceholders = Set([
```

Comment:
have you considered using an enum instead?
```
enum Placeholders: String, Decodable {
   case number
   case date
   case time
   ...
}
```

Reply:
And making that the type on `LocalizedString`? I thought about that, but went with this option since I think it gives better error messages than just the "JSON decoding failed" error.